src/palimpzest/sets.py

```diff
@@ -42,7 +42,7 @@ def __init__(
         limit: int | None = None,
         cardinality: Cardinality = Cardinality.ONE_TO_ONE,
         depends_on: list[str] | None = None,
-        nocache: bool = False,
+        cache: bool = True,
```

Comment:
TODO: It looks like we previously had a bug here because we specified `nocache = False` by default. You correctly flipped the logic, but what I think we actually want is: `cache: bool = False`